frontend/src/utils/util.ts

```diff
@@ -1,25 +1,31 @@
-export const convertPaginationToParams = (pagination) => {
+import { KeyLimitPagination } from 'staking/types/types';
+
+export const convertPaginationToParams = (
+  pagination?: KeyLimitPagination
+): string => {
   let result = '';
-  if (!pagination || (!pagination.key && !pagination.limit)) {
-    return '';
+  if (!pagination) {
+    return result;
   }
-  if (pagination.key !== null) {
+  if (pagination.key) {
     result += `pagination.key=${encodeURIComponent(pagination.key)}`;
-    if (pagination.limit !== null) {
-      result += `&pagination.limit=${pagination.limit}`;
-    }
-  } else {
-    if (pagination.limit !== null) {
-      result += `pagination.limit=${pagination.limit}`;
-    }
   }
-
+  if (pagination.limit) {
+    result += `pagination.limit=${pagination.limit}`;
+  }
   return result;
 };
 
-export const convertPaginationToParamsOffset = (pagination) => {
+export const convertPaginationToParamsOffset = (pagination: {
+  offset: number | null | undefined;
+  limit: number | null | undefined;
+}): string => {
   let result = '';
-  if (!pagination || (!pagination.key && !pagination.limit)) {
+  if (
+    pagination === undefined ||
+    (pagination?.offset === null && pagination?.limit === null) ||
+    (pagination?.offset === undefined && pagination?.limit === undefined)
```

Comment:
if (pagination?.offset == null && pagination?.limit == null) I think this should be enough to handle the case. Please test it once to make sure.

Reply:
resolved it further